risc0/zkvm/src/host/client/prove/mod.rs

```diff
@@ -139,8 +167,34 @@ impl ProverOpts {
         Self {
             hashfn: "sha-256".to_string(),
             prove_guest_errors: false,
+            receipt_format: ReceiptFormat::Composite,
         }
     }
+
+    /// Choose the prover that enables succinct receipts.
+    pub fn succinct() -> Self {
+        Self {
+            hashfn: "poseidon2".to_string(),
+            prove_guest_errors: false,
+            receipt_format: ReceiptFormat::Succinct,
+        }
+    }
+
+    /// Choose the prover that enables compact, snark receipts, only supported for x86_64 linux
+    #[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+    pub fn compact() -> Self {
+        Self {
+            hashfn: "poseidon2".to_string(),
+            prove_guest_errors: false,
+            receipt_format: ReceiptFormat::Compact,
+        }
+    }
+
+    /// Choose the prover that enables compact, snark receipts, only supported for x86_64 linux
+    #[cfg(not(all(target_arch = "x86_64", target_os = "linux")))]
+    pub fn compact() -> Self {
+        unimplemented!("compact receipts are only supported on x86_64 linux")
+    }
```

Comment:
Not sure about this, I'm thinking it's OK to fail later when it tries to run the docker image.

Reply:
Another thought: how about we catch it during proof generation so if this ProverOpts gets passed to ProveSession, I can do a check that bails earlier all of the receipts are generated?

Reply:
oh, the compact stuff will be in a later PR so I'll just drop the `unimplemented` for this PR